### PR TITLE
add reusable workflow templates for static sites

### PR DESCRIPTION
**Pull request by btylerburton**
related to:
  - https://github.com/GSA/data.gov/issues/3513

### CODE REVIEW

File: .github/workflows/static-site-qa-template.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Static Site Template
+
+on:   # yamllint disable-line rule:truthy
+  workflow_call:
+
+jobs:
+  qa:
+    name: Build Job
```
Comment (jbrown-xentity) [inline .github/workflows/static-site-qa-template.yml]: Can we give these different names? Something with qa...

**Review by jbrown-xentity — COMMENTED** [review]
With the following changes, should be good to go!

File: .github/workflows/static-site-qa-template.yml (hunk 1 above)
Comment (btylerburton) [reply .github/workflows/static-site-qa-template.yml]: i agree. `linkcheck`?

Comment (btylerburton) [reply .github/workflows/static-site-qa-template.yml]: I misinterpreted your comment. Updated the names. Give them a look.

**Review by nickumia-reisys — APPROVED** [review]
⚡

**Pull request merged**